src/plugins/data/public/ui/dataset_selector/configurator.tsx

```diff
@@ -41,8 +41,9 @@ export const Configurator = ({
   const languages = type?.supportedLanguages(baseDataset) || [];
 
   const [dataset, setDataset] = useState<Dataset>(baseDataset);
-  const [timeFields, setTimeFields] = useState<DatasetField[]>();
+  const [timeFields, setTimeFields] = useState<DatasetField[]>([]);
   const [timeFieldName, setTimeFieldName] = useState<string | undefined>(dataset.timeFieldName);
```

Comment:
oh i think i see why the issue above you mentioned. it is a string here but it's not a string it's a field object